sdk/cosmos/azure-cosmos/azure/cosmos/aio/container.py

```diff
@@ -199,25 +209,27 @@ async def create_item(
     @distributed_trace_async
     async def read_item(
             self,
-            item,  # type: Union[str, Dict[str, Any]]
-            partition_key,  # type: Any
-            **kwargs  # type: Any
+            item: Union[str, Dict[str, Any]],
+            partition_key: Union[str, int, float, bool],
+            **kwargs: Any
     ) -> Dict[str, Any]:
         """Get the item identified by `item`.
 
         :param item: The ID (name) or dict representing item to retrieve.
+        :type item: Union[str, Dict[str, Any]]
         :param partition_key: Partition key for the item to retrieve.
+        :type partition_key: Union[str, int, float, bool]
         :keyword str session_token: Token for use with Session consistency.
-        :keyword dict[str,str] initial_headers: Initial headers to be sent as part of the request.
-        :keyword Callable response_hook: A callable invoked with the response metadata.
+        :keyword Dict[str, str] initial_headers: Initial headers to be sent as part of the request.
+        :keyword response_hook: A callable invoked with the response metadata.
+        :paramtype response_hook: Callable[[Dict[str, str], Dict[str, Any]], None]
         **Provisional** keyword argument max_integrated_cache_staleness_in_ms
```

Comment:
Not relevant to this PR - but we should really try to find a better name for this before it releases.
Lastly, I don't think this notation is going to render properly - so we should double check the sphinx docs.